src/ovp_pipeline/source_signals/substack.py

```
@@ -0,0 +1,109 @@
+    r"^https?://(?P<handle>[\w-]+)\.substack\.com/(?:p/[\w-]+/?)?$"
+    r"|^https?://substack\.com/p/(?P<post_slug>[\w-]+)/?$"
+)
+
+
+@dataclass(frozen=True, slots=True)
+class SubstackSignalProvider:
+    """Stubbed provider — wires interface, defers implementation."""
+
+    name: str = "substack_authority"
+
+    def applies(self, source_url: str, frontmatter: dict[str, Any]) -> bool:
```

Ruff (0.15.12)

[warning] Unused method argument: `frontmatter`

(ARG002)